src/events/ready/01registerCommands.js

```diff
@@ -60,6 +60,18 @@ module.exports = async (client) => {
         console.log(`👍 Registered command "${name}."`);
       }
     }
+
+    // remove commands that no longer exist
+    for(const command of applicationCommands.cache) {
+      // command[0]: id
+      // command[1]: ApplicationCommand
+      if (!localCommands.find(c => c.name === command[1].name)) {
+        // server command does not exist here anymore
+        await applicationCommands.delete(command[1].id);
+          console.log(`🗑 Deleted command "${command[1].name}".`);
+      }
+    }
+
```

Comment:
Alternative (double check that this works):
```suggestion
    const legitCommands = new Set(localCommands.map(c => c.name));
    for(const command of applicationCommands.cache) {
      const id = command[0];
      const appCommand = command[1];
      if (!legitCommands.has(appCommand.name)) {
        // server command does not exist here anymore
        await applicationCommands.delete(appCommand.id);
          console.log(`🗑 Deleted command "${appCommand.name}".`);
      }
    }

```